enhancements/machine-api/aws-imds-v2-support.md

```diff
@@ -0,0 +1,276 @@
+    ...
+    // MetadataServiceOptions allows users to configure instance metadata service interaction options.
+    // If nothing specified, default AWS IMDS settings will be applied.
+    // https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_InstanceMetadataOptionsRequest.html
+    // +optional
+    MetadataServiceOptions *MetadataServiceOptions `json:"metadataServiceOptions,omitempty"`
+}
+```
+
+```go
+
+type InstanceMetadataHTTPTokens string
+
+const (
+    // InstanceMetadataHTTPTokensRequired enforces sending of a signed token header with any instance metadata retrieval (GET) requests.
+    // Enforces IMDSv2 usage.
+    InstanceMetadataHTTPTokensRequired = "required"
+    // InstanceMetadataHTTPTokensOptional allows IMDSv1 usage along with IMDSv2
+    InstanceMetadataHTTPTokensOptional = "optional"
+)
+
+// MetadataServiceOptions defines the options available to a user when configuring
+// Instance Metadata Service (IMDS) Options.
+type MetadataServiceOptions struct {
+    // HTTPTokens determines whether or not the host requires the use of authentication when interacting with the metadata service.
+    // When using authentication, this enforces v2 interaction method (IMDSv2) with the metadata service.
+    // Default: optional
+    // +kubebuilder:validation:Enum=required;optional
+    // +optional
+    HTTPTokens InstanceMetadataHTTPTokens `json:"httpTokens,omitempty"`
+    // HTTPPutResponseHopLimit is the desired HTTP PUT response hop limit for instance metadata requests.
+	// By default, the response to PUT requests has a response hop limit (time to live, TTL) of 1 at the IP protocol level.
+	// For example, you might need a larger hop limit for backward compatibility with container services running on the instance.
+    // The value may be anywhere between 1 and 64. When omitted, the default value is 1.
+    // + Pointer there because AWSMachineProviderConfig is a RawExtension within Machine resource.
+    // + OpenAPI validation does not apply there.
+    // +kubebuilder:validation:Minimum=1
+    // +kubebuilder:validation:Maximum=64
+    // +optional
+    HTTPPutResponseHopLimit *int32 `json:"httpPutResponseHopLimit,omitempty"`
+}
+```
+
+### Implementation Details/Notes/Constraints
+
+#### Install Config changes
+
+For supporting IMDSv2 during installation we could extend install-config with aws specific `metadataServiceOptions`
+section in the installer configuration.
+
+Example (origin is the [aws installation doc](https://docs.openshift.com/container-platform/4.10/installing/installing_aws/installing-aws-customizations.html#installation-aws-config-yaml_installing-aws-customizations)):
+
+```yaml
+apiVersion: v1
+baseDomain: example.com 
+controlPlane:   
+  hyperthreading: Enabled 
+  name: master
+  platform:
+    aws:
+      zones:
+      - us-west-2a
+      - us-west-2b
+      rootVolume:
+        iops: 4000
+        size: 500
+        type: io1
+      metadataServiceOptions: # proposed section
```

Comment:
```suggestion
      metadataService: # proposed section
```
For the install-config, I think we should drop `Options` from the yaml key. Everything in the install-config is options.